server/pkg/scene/sceneops/plugin_migrator.go

```diff
@@ -228,11 +175,72 @@ func (s *PluginMigrator) MigratePlugins(ctx context.Context, sc *scene.Scene, ol
 		Scene:             sc,
 		Layers:            modifiedLayers,
 		Properties:        properties,
-		RemovedLayers:     removedLayers,
+		RemovedLayers:     removedLayers.Layers(),
 		RemovedProperties: removedPropertyIDs,
 	}, nil
 }
 
+func isInvalidLayer(llp *layer.PluginID, lle *layer.PluginExtensionID, oldPlugin, newPlugin *plugin.Plugin) bool {
+	// a layer considered invalid if it is related to the old plugin
+	// and if the new plugin version does not support the extension
+	return llp != nil && lle != nil && (*llp).Equal(oldPlugin.ID()) && newPlugin.Extension(*lle) == nil
+}
+
```

Comment:
```suggestion
```

Layer's pluginID is always `reearth`, so it is no possible to change the plugin and extension ID of layers on plugin update